### PR TITLE
Tei tree updater

### PR DESCRIPTION
**Pull request by alicefuzier**
Part of https://github.com/wellcomecollection/catalogue-pipeline/issues/1536
Thi also adds builds for lambdas in buildkite cause I don't think the tests were running

### CODE REVIEW

File: tei_adapter/tei_updater/src/tei_updater.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+def diff_trees(old_tree, new_tree):
+    diff = DeepDiff(old_tree, new_tree, view="tree")
+    values_changed = diff.pop("values_changed", [])
+    items_added = diff.pop("dictionary_item_added", [])
+    items_removed = diff.pop("dictionary_item_removed", [])
+    assert len(diff.keys()) == 0
+    messages = []
+    # assert that the diff only contains the three keys above
+    if values_changed:
```
Comment (kenoir) [inline tei_adapter/tei_updater/src/tei_updater.py:52]:
```suggestion

    # assert that the diff only contains the three keys above
    assert len(diff.keys()) == 0
    
    messages = []

    if values_changed:
```

Comment (alicefuzier) [reply tei_adapter/tei_updater/src/tei_updater.py:52]: ah thanks

File: tei_adapter/tei_updater/src/test_tei_updater.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+def test_tree_does_not_exist(
+    mock_sns_client, test_topic_arn, get_test_topic_messages, session
+):
+    bucket = "bukkit"
+    key = "tree.json"
+    mock_s3_client = boto3.client("s3", region_name="us-east-1")
+    mock_s3_client.create_bucket(Bucket=bucket)
+    with mock.patch.dict(
+        os.environ,
+        {
```
Comment (kenoir) [inline tei_adapter/tei_updater/src/test_tei_updater.py:42]:
```suggestion
    mock_s3_client = boto3.client("s3", region_name="us-east-1")
    mock_s3_client.create_bucket(Bucket=bucket)
    
    with mock.patch.dict(
        os.environ,
        {
```

File: tei_adapter/tei_updater/src/test_tei_updater.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            "GITHUB_API_URL": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/trees/master?recursive=true",
+        },
+    ):
+        main({}, s3_client=mock_s3_client, sns_client=mock_sns_client, session=session)
+    messages = get_test_topic_messages()
+    assert len(list(messages)) == 653
+    content_object = mock_s3_client.get_object(Bucket=bucket, Key=key)
+    body = content_object["Body"].read()
+    saved_tree = json.loads(body)
+    assert len(saved_tree.keys()) == 653
```
Comment (kenoir) [inline tei_adapter/tei_updater/src/test_tei_updater.py:55]:
```suggestion
        },
    ):
    
    main({}, s3_client=mock_s3_client, sns_client=mock_sns_client, session=session)
    
    messages = get_test_topic_messages()
    assert len(list(messages)) == 653
    
    content_object = mock_s3_client.get_object(Bucket=bucket, Key=key)
    body = content_object["Body"].read()
    saved_tree = json.loads(body)
    assert len(saved_tree.keys()) == 653
```

Comment (kenoir) [reply tei_adapter/tei_updater/src/test_tei_updater.py:55]: You might also want to add some comments to these assertions to give a better idea of what's failed when it does? Maybe pytest also offers some nicer constructs to fail a test?

Comment (alicefuzier) [reply tei_adapter/tei_updater/src/test_tei_updater.py:55]: yeah, will add a comment. Those numbers come from the recording of the github api. Not sure about more elegant assertions though

**Review by kenoir — APPROVED** [review]
Looks good. Small point the formatting of some of the files is a bit dense - I'd prefer more whitespace :)

**Pull request merged**